src/Expression/JsonExpressionBuilder.php

```
@@ -51,6 +51,10 @@ public function __construct(private readonly QueryBuilderInterface $queryBuilder
      */
     public function build(ExpressionInterface $expression, array &$params = []): string
     {
+        if (!$expression instanceof JsonExpression) {
+            throw new InvalidArgumentException('JsonExpressionBuilder could only be used with JsonExpression.');
```

Should we throw `InvalidArgumentException` or `InvalidConfigException` here?